### ISSUE TITLE
Add the features like Keras

### ISSUE THREAD

**Issue by rose-jinyang** [pytorch/pytorch#58343]
## 🚀 Feature

1. The following layers are from Keras but PyTorch does NOT have these layers yet.

```
Conv2D(..., padding='same')
Conv2DTranspose(..., padding='same')
DepthwiseConv2D(..., padding='same')
SeparableConv2D(..., padding='same')

```

2.  Why does a PyTorch layer require an input channel surely?
    Keras does not require this and determines this value from a previous layer.
    Is it impossible in PyTorch?     

## Motivation

I think that PyTorch has many strong points in customizing the network training workflow.
So far, Tensorflow backend Keras has been used widely but many developers have several troubles in implementing several custom losses and so on.
So I am going to use PyTorch from now but I have some troubles in implementing Keras layers like Conv2D, Conv2DTranspose with 'same' padding in PyTorch too.
Also when composing a network in PyTorch, it is hard to enumerate an input channel value in each layer.
This results in many difficulties in implementing a DNN in PyTorch.

**Comment by mariosasko** [comment]
1. For odd kernel sizes with default stride and dilation, to achieve the "same" padding, the formula is: `kernel_size//2`. For even kernel sizes, check the [`F.pad`](https://pytorch.org/docs/1.8.0/nn.functional.html?highlight=pad#torch.nn.functional.pad) function in the docs. It shouldn't be too hard to find online implementations of the conv layers that are missing in PyTorch (PyTorch Forums and Stack Overflow are good places to start).

2. From torch 1.8 onwards you can use a [`LazyConv2D`](https://pytorch.org/docs/1.8.0/generated/torch.nn.LazyConv2d.html?highlight=lazy#torch.nn.LazyConv2d) module which will infer the number of input channels for you.
